code/common/subscriptions.q

```diff
@@ -78,18 +80,25 @@ subscribe:{[tabs;instrs;setschema;replaylog;proc]
 				.lg.o[`subscribe;"using the .sub.replayupd function as not replaying all tables or instruments"];
 				@[`.;`upd;:;.sub.replayupd[origupd;subtabs;instrs]]];
 			/-the second element of details is a pair (log count;logfile)
-			@[{-11!x;};details 1;{.lg.e[`subscribe;"could not replay the log file: ", x]}];
+			/-or list of pairs for stp ((count 1;log 1);(count 2;log 2);..)
+			d:$[0=type details[1;0];details 1;enlist details 1];
+			{[d]@[{-11!x;};d;{.lg.e[`subscribe;"could not replay the log file: ", x]}]}each d;
 			/-reset the upd function back to original upd
 			@[`.;`upd;:;origupd]];
-		if[replaylog&null details[1;1];
+		if[replaylog&null raze[details 1]1;
 			.lg.e[`subscribe;"replaylog set to true but TP not using log file"]];
 		/-insert the details into the SUBSCRIPTIONS table
 		.lg.o[`subscribe;"subscription successful"];
 		updatesubscriptions[proc;;instrs]each subtabs];
-		/-return the names of the tables that have been subscribed for and
-		/-the date from the name of thr tickerplant log file (assuming the tp log has a name like `: sym2014.01.01
-		/-plus .u.i and .u.icounts if existing on TP - details[1;0] is .u.i, details[2] is .u.icounts (or null)
-		(`subtables`tplogdate!(details[0;;0];(first "D" $ -10 sublist string last details 1)^logdate)),{(where 101 = type each x)_x}(`i`icounts`d)!(details[1;0];details[2];details[3])
+	/-return the names of the tables that have been subscribed for and
+	/-the date from the name of the tickerplant log file (assuming the tp log has a name like `: sym2014.01.01
+	/-plus .u.i and .u.icounts if existing on TP - details[1;0] is .u.i, details[2] is .u.icounts (or null)
+	if[`tickerplant=proc`proctype;
+		:(`subtables`tplogdate!(details[0;;0];(first "D" $ -10 sublist string last details 1)^logdate)),{(where 101 = type each x)_x}(`i`icounts`d)!(details[1;0];details[2];details[3])];
+	if[`segmentedtickerplant=proc`proctype;
+		retdic:(`logdir`subtables)!(`$getenv[`KDBSTPLOG];details[0;;0]);
+		retdic,:{(where 101 = type each x)_x}(`i`icounts`d`tplogdate)!(details[1;];details[2];details[3];("D"$8#-12 sublist string d1 first where not null d1:details[1;;1])^logdate);
```

Comment:
When getting the tplog date at the end here, -12 is not quite right. It should be

- -8 if in "none" or "tabular" mode e.g. stp1_20200807, trades_20200807
- -14 if in "periodic" or "tabperiod" e.g. periodic20200807050000
- more complicated in "custom" since it'll depend on the table